internal/controller/apiextensions/usage/reconciler.go

```diff
@@ -207,30 +209,23 @@ type Reconciler struct {
 
 	usage usageResource
 
-	log    logging.Logger
-	record event.Recorder
+	log        logging.Logger
+	record     event.Recorder
+	conditions conditions.Manager
 
 	pollInterval time.Duration
 }
 
 // Reconcile a Usage resource by resolving its selectors, defining ownership
 // relationship, adding a finalizer and handling proper deletion.
-func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) { //nolint:gocognit // Reconcilers are typically complex.
```

Comment:
Yes this is scope-creep but I wanted to get the project's take on moving to typed reconciled methods? It is a trivial change and starts us down the path of better pre/post-ample reconciliation logic that is outside of the typed reconciler.

Reply:
reverted and opened https://github.com/crossplane/crossplane/issues/6444